src/d3a/models/state.py

```diff
@@ -48,16 +48,16 @@ def __init__(self):
         self._energy_production_forecast_kWh = {}
 
     def get_energy_production_forecast_kWh(self, time_slot, default_value=None):
-        assert self._energy_production_forecast_kWh[time_slot] >= -FLOATING_POINT_TOLERANCE
+        assert self._energy_production_forecast_kWh.get(time_slot, 0.) >= -FLOATING_POINT_TOLERANCE
```

Comment:
Can you elaborate a bit on why this is needed ? Reason why I am asking is because this might mask some real problems with the code.

Reply:
https://github.com/gridsingularity/d3a/pull/1021/commits/811f96b603fb3f0a1a8cf9232f9a53ce5f77d03d